xbmc/cores/VideoPlayer/DVDCodecs/Video/VAAPI.cpp

```diff
@@ -123,14 +128,44 @@ bool CVAAPIContext::EnsureContext(CVAAPIContext **ctx, CDecoder *decoder)
   return true;
 }
 
+namespace
+{
+int find_open_render_node_fd()
```

Comment:
So, there are two parts to VAAPI usage:
1) Hardware video decoding
2) Presentation, using vaPutSurface.

(1) doesn't require that the GPU being used for decoding is the same that will be used for presentation; the underlying dma-buf mechanism makes this transparent¹. Indeed, if the non-display GPU happens to have better decoding hardware - for example, it supports H265 whereas the display GPU doesn't - it would be preferable to use the non-display GPU rather than fall back to software decoding.

This could be a sensible extension to the current code - open *each* DRM rendernode and use the most featureful one. A second extension would be to open each rendernode and potentially use both when decoding multiple streams - for picture-in-picture decoding, for example.

¹: It's slightly preferable to decode on the GPU you'll be displaying on, as that saves a GPU→GPU memory copy. They're reasonably fast, though; certainly they won't blow a 60fps budget.